src/main/java/org/wildfly/security/auth/realm/ldap/UserPasswordCredentialLoader.java

```diff
@@ -113,8 +117,9 @@ public SupportLevel getCredentialAcquireSupport(final Class<? extends Credential
 
         @Override
         public SupportLevel getEvidenceVerifySupport(final Class<? extends Evidence> evidenceType, final String algorithmName) throws RealmUnavailableException {
-            // If it is obtainable, it is verifiable.
-            return getCredentialAcquireSupport(PasswordCredential.class, null);
+            // If we can acquire PasswordCredential and it suppport provided evidence, we can verify.
+            if ( ! PasswordCredential.canVerifyEvidence(evidenceType, algorithmName)) return SupportLevel.UNSUPPORTED;
+            return getCredentialAcquireSupport(PasswordCredential.class, algorithmName);
```

Comment:
as discussed: it is responsibility of PasswordCredential to do the verification, it should also decide if it can verify some evidence